firehose-template.yaml

```diff
@@ -308,10 +314,10 @@ Resources:
         EndpointConfiguration:
           Name: New Relic
           Url: !FindInMap [NewRelicDatacenterMap, Datacenter, !Ref NewRelicRegion]
-          AccessKey: !If 
-            - ShouldCreateSecret
-            - !Sub '{{resolve:secretsmanager:${NewRelicLogsLicenseKeySecret}:SecretString:LicenseKey}}'
-            - !Ref LicenseKey
+          AccessKey: !If [ShouldCreateSecret, !Ref "AWS::NoValue",!Ref LicenseKey]
+        SecretsManagerConfiguration:
```

Comment:
@hrai-nr if ShouldCreateSecret is false then does it actually create secret manager but kept disabled?